eureka2-server/src/main/java/com/netflix/eureka2/server/service/SelfRegistrationService.java

```diff
@@ -86,6 +88,8 @@ public void call(Throwable throwable) {
         return resolver.resolve();
     }
 
+    public abstract void cleanUpResources();
```

Comment:
the typical pattern for shutdown overrides are 
void shutdown() {
   // do child shutdown
   super.shutdown().
}

In this case we want the cleanUp action to happen after the super.shutdown(), so this is a safer pattern.